pending_txs/pending_txs.go

```diff
@@ -43,23 +43,11 @@ func (apt *AccountPendingTxs) Add(layer types.LayerID, txs ...*types.Transaction
 	apt.mu.Unlock()
 }
 
-func (apt *AccountPendingTxs) Remove(accepted, rejected []*types.Transaction, layer types.LayerID) {
+func (apt *AccountPendingTxs) Remove(accepted []*types.Transaction, layer types.LayerID) {
 	apt.mu.Lock()
 	for _, tx := range accepted {
 		delete(apt.PendingTxs, tx.AccountNonce)
 	}
-	for _, tx := range rejected {
-		existing, found := apt.PendingTxs[tx.AccountNonce]
-		if found {
-			if existing[tx.Id()].HighestLayerIncludedIn > layer {
-				continue
-			}
-			delete(existing, tx.Id())
-			if len(existing) == 0 {
-				delete(apt.PendingTxs, tx.AccountNonce)
-			}
-		}
-	}
```

Comment:
This logic was here for a reason - I don't understand why it was removed.

It's also not done cleanly - the property `HighestLayerIncludedIn` is only ever used in this code, so if it's removed the property and the code that updates it should also be removed.